### DISCUSSION TITLE
Prevent quote marks in string formatted by `FailWith`

### DISCUSSION THREAD

**Discussion by lonix1** [fluentassertions/fluentassertions#1851] [Q&A]
In my custom assertion:
````cs
FailWith("foo.....{0}....bar.", something)
````

If `something` is a string, then the result is
> foo......"hello".......bar.

So it adds quotation marks. (It doesn't do that for other types, like ints.)

*Does someone know if it's possible to avoid that?*

My workaround is to use an interpolated string instead `$"foo....{something}......bar."`, but that's unfortunate because FA's string formatting is much cleaner.

**Comment by jnyrup** [comment]
Seems this also hit MS recently 

https://github.com/dotnet/runtime/pull/65730

**Reply by lonix1** [reply]
I didn't know MS uses FA internally... very cool! You and @dennisdoomen must be very proud, congrats!

**Reply by dennisdoomen** [reply]
I didn't know that either. That's very very cool.

**Reply by jnyrup** [reply]
https://www.nuget.org/packages/fluentassertions lists the most starred GitHub consumers under the "Used By" tab.
![image](https://user-images.githubusercontent.com/919634/158647594-011267d8-7e0f-4230-b8e7-33d89f32fa08.png)
